website/docs/r/serviceendpoint_azurerm.html.markdown

```diff
@@ -110,7 +205,7 @@ The following arguments are supported:
 A `credentials` block supports the following:
 
 - `serviceprincipalid` - (Required) The service principal application Id
-- `serviceprincipalkey` - (Required) The service principal secret.
+- `serviceprincipalkey` - (Optional) The service principal secret. This not required if `azurerm_service_endpoint_type` is set to `WorkloadIdentityFederation`.
```

Comment:
Not only not required, but not permitted (server would reject such calls).

Reply:
This field has to cater for ServicePrincipal too. If they choose the WorkloadIdentityFederation authenticaiton scheme, it will ignore this field and not pass it to the API.